evalml/pipelines/components/transformers/preprocessing/delayed_feature_transformer.py

```diff
@@ -105,7 +105,12 @@ def fit(self, X, y=None):
 
         Returns:
             self
+
+        Raises:
+            ValueError: if self.date_index is None
```

Comment:
I guess it is accurate because the `date_index` is ignored. It won't be ignored in #3028 .

The problem with checking in the init is that the default value of `date_index` is that `DelayedFeatureTransformer()` will raise an exception and that will cause `AutoMLSearch` to error out until https://github.com/alteryx/evalml/issues/1637 is done.